pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -570,23 +571,113 @@ type CloudConfig struct {
 		//issue body.
 		DisableStrictZoneCheck bool
 	}
+	// [ServiceOverride "1"]
+	//  Service = s3
+	//  Region = region1
+	//  URL = https://s3.foo.bar
+	//  SigningRegion = signing_region
+	//  SigningMethod = signing_method
+	//
+	//  [ServiceOverride "2"]
+	//     Service = ec2
+	//     Region = region2
+	//     URL = https://ec2.foo.bar
+	//     SigningRegion = signing_region
+	//     SigningMethod = signing_method
+	ServiceOverride map[string]*struct {
+		Service       string
+		Region        string
+		URL           string
+		SigningRegion string
+		SigningMethod string
```

Comment:
One more request before this goes, in, could you add `SigningName string` field?  It doesn't need to be a documented field, but it will really help AWS out with testing. An [empty string is value is fine](https://github.com/aws/aws-sdk-go/blob/173fcf0a78d6bc6c1668a416cdef0032c2a0efab/private/model/api/api.go#L318-L322), the SDK will just use the correct default.

Reply:
done